core/lib/spree/testing_support/event_helpers.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Spree
+  module TestingSupport
+    module EventHelpers
+      def perform_subscribers(only: nil, except: nil)
+        if only && except
+          raise ArgumentError, <<~ERROR.strip
+            You cannot pass both `:only` and `:except` to `perform_subscribers`!
+          ERROR
+        end
+
+        _with_subscriber_registry_override do
+          Spree::Event.deactivate_all_subscribers
+
+          registry = Spree::Config.events.subscriber_registry
+
+          if only # perform_subscribers(only: [Subscriber1, Subscriber2]) { ... }
+            Array(only).each { |subscriber| registry.activate_subscriber(subscriber) }
+          elsif except # perform_subscribers(except: [Subscriber1, Subscriber2]) { ... }
+            registry.activate_all_subscribers
+            Array(except).each { |subscriber| registry.deactivate_subscriber(subscriber) }
+          else # perform_subscribers { ... }
+            registry.activate_all_subscribers
+          end
+
+          yield
+        end
+      end
+
+      private
+
+      def _deactivate_all_subscribers
```

Comment:
Up to you, but if the concern is that you want to avoid polluting the test context with private methods, what about extracting the helper's whole functionality to an object or class invoked by the helper itself? That's usually my approach for mixins that don't require any access to the context of where they're being mixed in.

You and I know why these methods have prepended underscores, but it seems like something non-obvious to the average developer who might look at this. Giving them underscores also doesn't make them any less callable or visible in a test context either.

It's a tradeoff, since you're adding some indirection, but the whole implementation could still live in-line in this file, so I don't think it would add any confusion.

Not blocking though.